src/crawlee/basic_crawler/_basic_crawler.py

```diff
@@ -88,14 +88,56 @@ class BasicCrawlerOptions(TypedDict, Generic[TCrawlingContext]):
-    crawling of websites.
-
-    `BasicCrawler` is a low-level tool that requires the user to implement the page download and data extraction
-    functionality themselves. If we want a crawler that already facilitates this functionality, we should consider using
-    one of its subclasses.
+    """A versatile web crawler for parallel URL fetching with extensive features for web scraping.
+
+    BasicCrawler is a highly customizable and efficient web crawling solution that provides:
+    - Automatic scaling based on available system resources and target website's requirements
+    - Smart request routing and handling of different types of URLs
+    - Session management for maintaining persistent connections
+    - Proxy integration and rotation for avoiding IP-based blocking
+    - Automatic retries for failed requests with customizable settings
+    - Built-in statistics tracking for monitoring crawler performance
+    - Event system for fine-grained control over the crawling process
+
+    Basic usage:
+    ```python
+    async def handle_request(context: Context) -> None:
+        url = context.request.url
+        response = await context.send_request()
+        # Process the response here
+    crawler = BasicCrawler(
+        request_handler=handle_request,
+    )
+    await crawler.run()
+    ```
+
+    Advanced features:
+    - Configure concurrency to control parallel processing
+    - Set up automatic retry mechanisms for failed or blocked requests
+    - Utilize session pools for efficient connection management
+    - Integrate proxy services for IP rotation
+    - Monitor detailed statistics about the crawling process
+
+    Args:
+        request_provider: The provider for requests to be crawled.
+        request_handler: The function to handle individual requests.
+        http_client: The HTTP client for making web requests.
+        concurrency_settings: Settings for controlling concurrency.
+        max_request_retries: Maximum retry attempts for failed requests. Defaults to 3.
+        max_requests_per_crawl: Maximum number of requests per crawl. No limit by default.
+        max_session_rotations: Maximum session rotations per request. Defaults to 10.
+        configuration: Crawler configuration settings.
+        request_handler_timeout: Request handler timeout. Defaults to 1 minute.
+        use_session_pool: Whether to use session pool. Defaults to True.
+        session_pool: Preconfigured session pool.
+        retry_on_blocked: Whether to retry on blocked requests. Defaults to True.
+        proxy_configuration: Proxy configuration.
+        statistics: Statistics object.
+        event_manager: Event manager.
+        configure_logging: Whether to configure logging. Defaults to True.
+        _context_pipeline: Internal context pipeline.
+        _additional_context_managers: Additional context managers.
+        _logger: Logger instance.
```

Comment:
Please remove `Defaults to...` as this is contained already in the function declaration. Again, this applies to all docstrings.